ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateSignatureProvider.fs

```diff
@@ -266,15 +266,29 @@ type FSharpGenerateSignatureBuilder() =
                 factory.CreateTypeMemberSignature(sourceString)
             | _ -> null
 
-        and createPrimaryConstructorSignature (typeName: string) (primaryConstructorDeclaration: IPrimaryConstructorDeclaration) : IFSharpTreeNode =
-            let signature =
-                match primaryConstructorDeclaration.ParameterPatterns with
-                | :? IUnitPat ->
-                    $"new: unit -> {typeName}"
-                | _ -> failwithf "todo"
+        and createPrimaryConstructorSignature (typeName: string) (primaryConstructorDeclaration: IPrimaryConstructorDeclaration) : ITreeNode seq =
+            match primaryConstructorDeclaration.ParameterPatterns with
+            | :? IUnitPat ->
+                factory.CreateTypeMemberSignature $"new: unit -> {typeName}"
+                :> ITreeNode
+                |> Seq.singleton 
+            | _ ->
```

Comment:
Is this check actually required? I'd expect getting `FcsSymbolUse` would work for empty primary constructors too.